infrastructure/templates/public-api/main.bicep

```diff
@@ -1,290 +1,206 @@
-@minLength(5)
-@maxLength(50)
-@description('Registry : Name of the azure container registry (must be globally unique)')
-param containerRegistryName string
+param postgreSqlSkuName string = 'Standard_B1ms'
 
-@description('Deploy the Container Registry if you are not using an existing registry')
-param deployRegistry bool
+@description('Database : Azure Database for PostgreSQL Storage Size in GB.')
+param postgreSqlStorageSizeGB int = 32
 
-//Container App Params -------------------------------------------------------------------
-@minLength(2)
-@maxLength(32)
-@description('Specifies the name of the container app.')
-param containerAppName string
+@description('Database : Azure Database for PostgreSQL Autogrow setting.')
+param postgreSqlAutoGrowStatus string = 'Disabled'
 
-@description('Specifies the name of the container app environment.')
-param containerAppEnvName string = 'publicapi'
+@description('Database : Firewall rules.')
+param postgreSqlFirewallRules array = []
 
-@description('Specifies the name of the log analytics workspace.')
-param containerAppLogAnalyticsName string = 'publicapi'
+@description('Container App : Select if you want to use a public dummy image to start the container app.')
+param useDummyImage bool = true
 
-@description('Container App : Specifies the container image to deploy from the registry <name>:<tag>.')
-param containerAppImageName string
+param now string = utcNow('u')
 
-@description('Specifies the container port.')
-param containerAppTargetPort int = 80
-
-@description('Select if you want to use a public dummy image to start the container app.')
-param useDummyImage bool
-
-
-//ServiceBus Queue Params -------------------------------------------------------------------
-@description('Name of the Service Bus namespace')
-param namespaceName string = 'processor'
-
-@description('Name of the Queue')
-param queueName string = 'Processorqueue'
+//---------------------------------------------------------------------------------------------------------------
+// Variables and created data.
+//---------------------------------------------------------------------------------------------------------------
+var project = 'ees-publicapi'
+var resourcePrefix = '${subscription}-${project}'
+var storageAccountFullName = '${subscription}saeescore'
+var keyVaultName = 'kv-ees-01'
+var keyVaultFullName = '${subscription}-${keyVaultName}'
+var containerAppName = 'api'
+var containerAppImageName = useDummyImage ? 'azuredocs/aci-helloworld' : 'real-container-image-name'
+var containerAppTargetPort = 80
+var rootFileShareFolderName = 'data'
+var containerRegistryName = 'eesacr'
+var databaseNames = ['publicapi']
+
+var tagValues = {
+  departmentName: 'Unknown'
+  environmentName: environmentName
+  solutionName: 'API'
+  subscriptionName: 'Unknown'
+  costCentre: 'Unknown'
+  serviceOwnerName: 'Unknown'
+  dateProvisioned: now
+  createdBy: 'Unknown'
+  deploymentRepo: 'N/A'
+  deploymentScript: 'main.bicep'
+}
```

Comment:
Well that wasn't easy... :joy: 

However, tada!

![image](https://github.com/dfe-analytical-services/explore-education-statistics/assets/12444316/d9dae9f9-40ba-42f4-a40c-a158ef6180e8)

These tag values are fed from the "EES Bicep Infrastructure Resource Tags" variable group, from its "resourceTags" variable, which I chose to supply as a single piece of JSON rather than passing around loads of individual key / values:

![image](https://github.com/dfe-analytical-services/explore-education-statistics/assets/12444316/61301e1a-eddc-4249-a88a-4f91186fa937)

These are supplied to the validate and deploy steps using the final `--parameters` line:

```
                az deployment group create \
                --resource-group ${{parameters.resourceGroupName}} \
                --template-file $(templateFile) \
                --parameters $(devParamFile) \
                --parameters resourceTags='$(resourceTags)'
```

Reply:
Note that I've gone with using a combination of one plainly visible Variable Group and one Variable Group that is linked to secrets from the appropriate Resource Group's Key Vault.  Because of this, I've had to switch to the ugly "name / value" format of the `variables` definition in `api-infrastructure-pipeline.yml` - this is the only way you can specify more than one variable group here - not a deliberate choice on my part!